# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1539806003 -7200
#      Wed Oct 17 21:53:23 2018 +0200
# Node ID 91711db98d8aee8055de296b90a2193dfb6bdae8
# Parent  4b661d535ddb3d983a55734a5644016b50338c4d
Fix misbehavior of axes position when "hold on" and labels are present (bug #48718)

* graphics.in.(h|.cc) (axes::properties::calc_tightbox): Force an update of the
  ticklength and the labels position before requesting the axes extent.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -4993,16 +4993,21 @@ axes::properties::init (void)
 */
 
 Matrix
 axes::properties::calc_tightbox (const Matrix& init_pos)
 {
   Matrix pos = init_pos;
   graphics_object go = gh_manager::get_object (get_parent ());
   Matrix parent_bb = go.get_properties ().get_boundingbox (true);
+
+  // FIXME: The layout should be clean at this stage and we should not have to
+  //        update ticks and labels positions here again. See bug #48718.
+  update_ticklength ();
+
   Matrix ext = get_extent (true, true);
   ext(1) = parent_bb(3) - ext(1) - ext(3);
   ext(0)++;
   ext(1)++;
   ext = convert_position (ext, "pixels", get_units (),
                           parent_bb.extract_n (0, 2, 1, 2));
   if (ext(0) < pos(0))
     {
